libs/libc/unistd/lib_pathconf.c

```diff
@@ -0,0 +1,129 @@
+ *     {MAX_INPUT}                        _PC_MAX_INPUT
+ *     {NAME_MAX}                         _PC_NAME_MAX
+ *     {PATH_MAX}                         _PC_PATH_MAX
+ *     {PIPE_BUF}                         _PC_PIPE_BUF
+ *     {POSIX2_SYMLINKS}                  _PC_2_SYMLINKS
+ *     {POSIX_ALLOC_SIZE_MIN}             _PC_ALLOC_SIZE_MIN
+ *     {POSIX_REC_INCR_XFER_SIZE}         _PC_REC_INCR_XFER_SIZE
+ *     {POSIX_REC_MAX_XFER_SIZE}          _PC_REC_MAX_XFER_SIZE
+ *     {POSIX_REC_MIN_XFER_SIZE}          _PC_REC_MIN_XFER_SIZE
+ *     {POSIX_REC_XFER_ALIGN}             _PC_REC_XFER_ALIGN
+ *     {SYMLINK_MAX}                      _PC_SYMLINK_MAX
+ *     _POSIX_CHOWN_RESTRICTED            _PC_CHOWN_RESTRICTED
+ *     _POSIX_NO_TRUNC                    _PC_NO_TRUNC
+ *     _POSIX_VDISABLE                    _PC_VDISABLE
+ *     _POSIX_ASYNC_IO                    _PC_ASYNC_IO
+ *     _POSIX_PRIO_IO                     _PC_PRIO_IO
+ *     _POSIX_SYNC_IO                     _PC_SYNC_IO
+ *
+ * Returned Value:
+ *   If name is an invalid value, both pathconf() and fpathconf() shall
+ *   return -1 and set errno to indicate the error.
+ *
+ *   If the variable corresponding to name has no limit for the path or file
+ *   descriptor, both pathconf() and fpathconf() shall return -1 without
+ *   changing errno. If the implementation needs to use path to determine the
+ *   value of name and the implementation does not support the association of
+ *   name with the file specified by path, or if the process did not have
+ *   appropriate privileges to query the file specified by path, or path does
+ *   not exist, pathconf() shall return -1 and set errno to indicate the
+ *   error.
+ *
+ *   If the implementation needs to use fildes to determine the value of name
+ *   and the implementation does not support the association of name with the
+ *   file specified by fildes, or if fildes is an invalid file descriptor,
+ *   fpathconf() shall return -1 and set errno to indicate the error.
+ *
+ *   Otherwise, pathconf() or fpathconf() shall return the current variable
+ *   value for the file or directory without changing errno. The value
+ *   returned shall not be more restrictive than the corresponding value
+ *   available to the application when it was compiled with the
+ *   implementation's <limits.h> or <unistd.h>.
+ *
+ *   If the variable corresponding to name is dependent on an unsupported
+ *   option, the results are unspecified.
+ *
+ ****************************************************************************/
+
+long fpathconf(int fildes, int name)
+{
+  /* NOTE:  The initialize implementation of this interface is very sparse.
+   * It was originally created to support only the functionality of libcxx
+   * but can be extended to support as much of the standard POSIX as is
+   * necessary.
+   */
+
+  switch (name)
+    {
+      case _PC_PATH_MAX:
+        return PATH_MAX;
+
+      default:
+        set_errno(ENOSYS);
```

Comment:
This will also set errno to `ENOSYS` for an invalid name, which is not the standard behavior.  However, I see that `sysconf` does the same thing as well with a comment explaining the situation, could we also add the same comment here for future references?

Reply:
Done.